docs/ChangeLog/20221126.md

```diff
@@ -19,6 +23,401 @@ The following keyboards have had their source moved within QMK:
 | handwired/hillside/52                | hillside/52                          |
 | maple_computing/christmas_tree/V2017 | maple_computing/christmas_tree/v2017 |
 
+### Keycodes :id=keycodes-overhaul-user-action
+
+QMK's keycodes got a very significant overhaul this breaking changes cycle, with the bulk of the work done by _@zvecr_ and _@fauxpark_ -- renaming, reordering, removing has been their focus in this area. In an attempt to standardise interoperation with host applications, keycode values now have strong versioning so that any connected application has confidence that the keys it thinks exist on the board actually match up with what's compiled in. These strongly-versioned keycode definitions are now published online and will not change, so tools that remap keycodes have a reference to work with. In future versions of QMK, any new or changed keycodes will result in a new version specification. See [API docs](api_docs.md#qmk-constants) for more information on the published versions if you're writing a tool to manage keycodes.
+
+In most cases user keymaps in the repository have already been updated to reflect the new naming scheme. In some cases user keymaps outside the repository may strike a missing keycode with the old name -- it's highly likely that the name had already been deprecated for some time, and should have been updated previously.
+
+See below for the full list of changesets.
+
+!> Keycode aliases have been put in place in most cases to cater for "old names" being mapped to "new names" -- the documentation already reflects all the new naming of keys.
+
+### Data-driven USB IDs Refactoring ([#18152](https://github.com/qmk/qmk_firmware/pull/18152)) :id=usb-ids-Refactoring
+
+QMK has decided to deprecate the specification of USB IDs inside `config.h` in favour of `info.json`, leaving data-driven as the only method to specify USB information. As per the deprecation schedule put forward last breaking changes cycle, USB information must be specified in `info.json` instead.
+
+Previously in `config.h`:
+```c
+#define VENDOR_ID    0x1234
+#define PRODUCT_ID   0x5678
+#define DEVICE_VER   0x0001
+#define MANUFACTURER Me
+#define PRODUCT      MyKeyboard
+```
+
+Replaced by `info.json`:
+```json
+{
+    "keyboard_name": "MyKeyboard",
+    "manufacturer": "Me",
+    "usb": {
+        "vid": "0x1234",
+        "pid": "0x5678",
+        "device_version": "0.0.1"
+    }
+}
+```
+
```

Comment:
Maybe?
````suggestion

### LED Indicator callback refactoring ([#14864](https://github.com/qmk/qmk_firmware/pull/18450)) :id=led-callback-refactor

_RGB Matrix_ and _LED Matrix_ Indicator display code was traditionally difficult to override in keymaps as they did not follow the standard pattern of `bool *_kb()` deferring to `bool *_user()` functions, allowing signalling to the higher level that processing had already been done.

This changes the standard callback model to allow for a base implementation to be provided by a keyboard, but also still allow for keymap-level overrides without needing to modify the keyboard's code.

The old keymap code went something like this:

```c
void rgb_matrix_indicators_user(void) {
    // keymap LED code
}
```

...but the new keymap code looks like this:
```c
bool rgb_matrix_indicators_user(void) {
    // keymap LED code
    return false;
}
```

Keyboard designers should now structure their keyboard-level routines like the following, in order to allow for keymap overrides:

```c
bool rgb_matrix_indicators_kb(void) {
    // Defer to the keymap if they want to override
    if (!rgb_matrix_indicators_user()) { return false; }

    // keyboard LED code
    return true;
}
```

````